src/fluid/fluid.cpp

```diff
@@ -319,6 +322,23 @@ std::shared_ptr<StateDescriptor> Initialize(ParameterInput *pin) {
   auto ReduceEn = [](MeshData<Real> *md) {
     return ReduceOneVar<Kokkos::Sum<Real>>(md, fluid_cons::energy::name(), 0);
   };
+  auto MaxDensitySN = [](MeshData<Real> *md) {
+    return ReduceOneVar<Kokkos::Max<Real>>(md, fluid_prim::density::name(), 0);
+  };
+  auto Mgain = [](MeshData<Real> *md) {
+    return ReduceInGain<Kokkos::Sum<Real, HostExecSpace>>(md, fluid_prim::density::name(),
+                                                          0);
+  };
+  auto Qgain = [](MeshData<Real> *md) {
+    return ReduceInGain<Kokkos::Sum<Real, HostExecSpace>>(
+               md, internal_variables::GcovHeat::name(), 0) -
+           ReduceInGain<Kokkos::Sum<Real, HostExecSpace>>(
+               md, internal_variables::GcovCool::name(), 0);
+  };
+
+  hst_vars.emplace_back(HistoryOutputVar(HstMax, MaxDensitySN, "maximum density SN"));
+  hst_vars.emplace_back(HistoryOutputVar(HstSum, Mgain, "Mgain"));
+  hst_vars.emplace_back(HistoryOutputVar(HstSum, Qgain, "total net heat"));
```

Comment:
These should not be on by default, as they'll interfere with, e.g.,  simulations by @AstroBarker and @brryan . A few possible solutions:
1. Add a flag in `fluid.cpp` that turns them on or off depedning on the input deck
2. Move them into the progenitor package so they won't be active for non-SN sims

I suggest (2) but (1) is fine if you prefer.